bin/github-review.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+    cat <<EOF
+This script verifies the deployed contracts on a chain ID for a given PR.
+
+USAGE
+    bash ./bin/github-review.sh <PR> <CHAIN_ID> <RPC_URL> <VERSION>
+
+ARGUMENTS
+    PR          The GitHub PR number
+    CHAIN_ID    The chain ID to verify
+    RPC_URL     The RPC URL to use for the chain ID
+    VERSION     The version of the contracts to verify
+
+EXAMPLES
+    bash ./bin/github-review.sh 123 1 https://rpc.ankr.com/eth 1.3.0
+    bash ./bin/github-review.sh 123 1 https://rpc.ankr.com/eth 1.4.1
+EOF
+}
+
+if [[ -n "$(git status --porcelain)" ]]; then
+    echo "ERROR: Dirty Git index, please commit all changes before continuing" 1>&2
+    exit 1
+fi
+if ! command -v gh &> /dev/null; then
+    echo "ERROR: Please install the 'gh' GitHub CLI" 1>&2
+    exit 1
+fi
+if ! command -v cast &> /dev/null; then
+    echo "ERROR: Please install the 'cast' tool included in the Foundry toolset" 1>&2
+    exit 1
+fi
+
+pr=0
+chainid=0
+rpc=0
+version=""
+availableVersion=("1.3.0" "1.4.1")
+case $# in
+    4)
+        if ! [[ $1 =~ ^[0-9]+$ ]]; then
+            echo "ERROR: $1 is not a valid GitHub PR number" 1>&2
+            usage
+            exit 1
+        fi
+        pr=$1
+        if ! [[ $2 =~ ^[0-9]+$ ]]; then
+            echo "ERROR: $2 is not a valid Chain ID number" 1>&2
+            usage
+            exit 1
+        fi
+        chainid="$(cast chain-id --rpc-url $3)"
+        if [[ $chainid != $2 ]]; then
+            echo "ERROR: RPC $3 doesn't match chain ID $2" 1>&2
+            usage
+            exit 1
+        fi
+        rpc=$3
+        if [[ ! " ${availableVersion[@]} " =~ " ${4} " ]]; then
+            echo "ERROR: $4 is not a valid version" 1>&2
+            usage
+            exit 1
+        fi
+        version=$4
+        ;;
+    *)
+        usage
+        exit 1
+        ;;
+esac
+
+echo "Verifying Deployment Asset"
+gh pr diff $pr --patch | git apply --include 'src/assets/**'
+
+# Getting default addresses
+oneThreeZeroFiles=("compatibility_fallback_handler" "create_call" "gnosis_safe_l2" "gnosis_safe" "multi_send_call_only" "multi_send" "proxy_factory" "sign_message_lib" "simulate_tx_accessor")
+oneFourOneFiles=("compatibility_fallback_handler" "create_call" "multi_send_call_only" "multi_send" "safe_l2" "safe_proxy_factory" "safe" "sign_message_lib" "simulate_tx_accessor")
+versionFiles=()
+versionAddresses=()
+
+if [[ $version == "1.3.0" ]]; then
+    versionFiles=("${oneThreeZeroFiles[@]}")
+elif [[ $version == "1.4.1" ]]; then
+    versionFiles=("${oneFourOneFiles[@]}")
+fi
+
+# Getting default addresses
+for file in "${versionFiles[@]}"; do
+    versionAddresses+=($(jq -r '.defaultAddress' "src/assets/v$version/$file.json"))
```

Comment:
One issue with this method is that it doesn't work for zkSync based networks or all deployments in v1.3.0 (there are two possible addresses there).